workshops/konami_code/README.md

```diff
@@ -0,0 +1,478 @@
+
+Here's how to do it:
+
+```js
+keys.push(e.key);
+```
+Try console logging the `keys` array inside the function. RUN the code and try pressing the keys. You'll see the keys getting pushed inside the array!
+
+**Note:** Make sure your preview window is focused when you are pressing the keys.
+
+We don't need the `keys` array to get bigger and bigger. We definitely need a limit, right? That limit will be the length of our `secretCode` array. Whenever the number of keys exceed the length of `secretCode` array, we will remove the first element of the `keys` array.
+
+Here's how you will do it:
+
+```js
+function keysDetector(e) {
+... // all the above code
+
+if (keys.length > secretCode.length) {
+		keys.shift()
+	}
+}
+```
+
+**Explanation:** Just like `Array.push()` there is also a method `Array.shift()` which removes the first element from an array! Here we implement the code such that, whenever the `keys` array's length exceeds the `secretCode` array's length, we remove its first element. So, in that way, the number of keys stored in our array is fixed.
+
+Try console logging the `keys` array now (inside the function) and you'll get a better understanding of the above code.
+
+And now, the time has come! We will compare both the arrays and whenever `keys === secretCode` we will do some magic!
+
+But wait a minute...
+
+![Image](https://media.giphy.com/media/RkKMFRHh71RvhCYUx6/giphy.gif)
+
+Unfortunately, we can't simply write `keys === secretCode` and get away with it :( The reason for this is as arrays are objects in JavaScript, the triple equals operator `(===)` only returns true if the arrays are the same reference.
+
+Example:
+```js
+const a = [1, 2, 3];
+const b = [1, 2, 3];
+
+a === a; // true
+a === b; // false
+```
+
+So, we'll need to go another way which is known as...  
+**`JSON.stringify()`**!
+
+The JSON.stringify() method will convert a JavaScript object or value to a JSON string. So, we will first convert both the arrays to a string and then, we are free to compare it!
+
+Here's how to do it:
+```js
+if (JSON.stringify(keys) === JSON.stringify(secretCode)) {
+  //do something or lets just console.log for now.
+  console.log('TADAAA!!');
+}
+```
+**Note:** This will also be included in the function. Except the event listener, nothing is outside the function.
+
+RUN the code and try typing the secret code which you have given (it is "hack" in my case). You'll surely see a TADAAA!! in the console whenever the arrays match each other.
+
+Okay, so now we are sure that our code works perfectly! So, instead of console logging we will make some pretty changes to our website.
+
+First of all, we'll make our `secret-div` visible. Then we'll change the `text` from 'Password Please' to 'Identity Verified'. And not to forget, we'll also render confetti on our page.
+
+**Challenge:** Can you do this all on your own?  
+**Note:** Do it inside the `if` statement and also remove the `console.log()`.  
+**Hints:**  
```

Comment:
```suggestion
**Note:** This will also be included in the function. Except for the event listener, nothing is outside the function.

RUN the code and try typing the secret code which you have given (it is "hack" in my case). You'll surely see a TADAAA!! in the console whenever the arrays match each other.

Okay, so now we are sure that our code works perfectly! So, instead of console logging, we will make some pretty changes to our website.

First of all, we'll make our `secret-div` visible. Then we'll change the `text` from 'Password Please' to 'Identity Verified'. And not to forget, we'll also render confetti on our page.

**Challenge:** Can you do this all on your own?  
**Note:** Do it inside the `if` statement and also remove the `console.log()`.  
**Hints:**  
```